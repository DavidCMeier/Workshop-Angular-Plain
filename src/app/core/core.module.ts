import { StoreModule } from "@ngrx/store";
import { EffectsModule } from "@ngrx/effects";
import * as fromStore from './store';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderComponent } from './components/header/header.component';
import { FooterComponent } from './components/footer/footer.component';
import { ShellComponent } from './containers/shell/shell.component';
import { MatToolbarModule } from "@angular/material/toolbar";
import { RouterModule } from "@angular/router";
import { MatSelectModule } from "@angular/material/select";
import { MatIconModule } from "@angular/material/icon";
import { MatButtonModule } from "@angular/material/button";
import { HTTP_INTERCEPTORS, HttpClientModule } from "@angular/common/http";
import { AuthInterceptor } from "./interceptors/auth-interceptor.service";
import { SharedModule } from "../shared/shared.module";
import { StoreRouterConnectingModule } from "@ngrx/router-store";



@NgModule({
  declarations: [
    HeaderComponent,
    FooterComponent,
    ShellComponent
  ],
  imports: [
    StoreModule.forFeature('core', fromStore.reducers),
    EffectsModule.forFeature(fromStore.effects),
    StoreRouterConnectingModule.forRoot(),
    CommonModule,
    MatToolbarModule,
    RouterModule,
    MatSelectModule,
    MatIconModule,
    MatButtonModule,
    HttpClientModule,
    SharedModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true }
  ],
})
export class CoreModule { }
