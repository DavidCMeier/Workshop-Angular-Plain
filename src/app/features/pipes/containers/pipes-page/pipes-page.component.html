<h1>Pipes</h1>

<!--<p>LowerCasePipe: {{'HOLA MUNDO' | lowercase}}</p>-->
<!--<p>UpperCasePipe: {{'hola mundo' | uppercase}}</p>-->
<!--<p>TitleCasePipe: {{'hola mundo' | titlecase}}</p>-->
<!--<p>JsonPipe: {{{name: 'David', age: 33} | json}}</p>-->
<!--<p>CurrencyPipe por defecto: {{ 1.39 | currency}}</p>-->
<!--<p>CurrencyPipe con Euro: {{ 1.39 | currency: 'EUR'}}</p>-->
<!--<p>DatePipe por defecto: {{'2022-03-14' | date}}</p>-->
<!--<p>DatePipe con fecha en formato dd/MM/yyyy: {{'2022-03-14' | date: 'dd/MM/YYYY'}}</p>-->

<!--<h2>Custom Pipe</h2>-->
<!--<p>Capitalize: {{'buenos días, ¿Cómo estáis?' | capitalize}}</p>-->
