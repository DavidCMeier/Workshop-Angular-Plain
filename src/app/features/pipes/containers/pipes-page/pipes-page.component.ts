import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-pipes-page',
  templateUrl: './pipes-page.component.html',
  styleUrls: ['./pipes-page.component.scss']
})
export class PipesPageComponent implements OnInit {

  constructor() { }

  ngOnInit(): void {
  }

}
