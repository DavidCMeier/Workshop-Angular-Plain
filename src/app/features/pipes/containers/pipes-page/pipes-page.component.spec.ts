import { ComponentFixture, TestBed } from '@angular/core/testing';

import { PipesPageComponent } from './pipes-page.component';
import { CapitalizePipe } from "../../../../shared/pipes/capitalize.pipe";

describe('PipesPageComponent', () => {
  let component: PipesPageComponent;
  let fixture: ComponentFixture<PipesPageComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PipesPageComponent, CapitalizePipe ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PipesPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it('should be pass the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
