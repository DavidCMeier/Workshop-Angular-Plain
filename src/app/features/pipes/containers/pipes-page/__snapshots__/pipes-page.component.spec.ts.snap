// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipesPageComponent should be pass the snapshot 1`] = `
<app-pipes-page
  __ngContext__={[Function LRootView]}
>
  <h1>
    Pipes
  </h1><p>
    LowerCasePipe: hola mundo
  </p><p>
    UpperCasePipe: HOLA MUNDO
  </p><p>
    TitleCasePipe: Hola Mundo
  </p><p>
    JsonPipe: {
    "name": "David",
    "age": 33
  }
  </p><p>
    CurrencyPipe por defecto: $1.39
  </p><p>
    CurrencyPipe con Euro: €1.39
  </p><p>
    DatePipe por defecto: Mar 14, 2022
  </p><p>
    DatePipe con fecha en formato dd/MM/yyyy: 14/03/2022
  </p><h2>
    Custom Pipe
  </h2><p>
    Capitalize: Buenos días, ¿Cómo estáis?
  </p>
</app-pipes-page>
`;
